MacPorts sets CMAKE_INSTALL_NAME_DIR, which is not correct for all shared libraries.
--- router/cmake/Plugin.cmake.orig	2019-08-14 07:25:47.000000000 -0700
+++ router/cmake/Plugin.cmake	2019-09-21 05:35:00.000000000 -0700
@@ -127,6 +127,8 @@
   IF(APPLE)
     SET_TARGET_PROPERTIES(${NAME} PROPERTIES
         LINK_FLAGS "-undefined dynamic_lookup")
+    SET_TARGET_PROPERTIES(${NAME} PROPERTIES
+        INSTALL_NAME_DIR ${CMAKE_INSTALL_PREFIX}/${_option_DESTINATION})
   ENDIF()
 
   SET_TARGET_PROPERTIES(${NAME} PROPERTIES
